### PR TITLE
Authenticate endpoint + cleanup

### PR DESCRIPTION
**Pull request by wallin**

### CODE REVIEW

**Comment by lluft** [comment]
yay `identify` support!

File: test/CastleTest.php [hunk 1]
```diff
@@ -20,66 +19,30 @@ public function testSetApiKey()
...
-
   public function testAuthenticate()
   {
-    Castle_RequestTransport::setResponse(200, '{}');
-    Castle::authenticate(1);
-    $this->assertRequest('post', '/authentications');
+    Castle_RequestTransport::setResponse(201, '{ "status": "approve" }');
+    $auth = Castle::authenticate(Array(
+      'user_id' => '1',
+      'name' => '$login.failed'
```
Comment (lluft) [inline test/CastleTest.php]: change this to `$login.succeeded` to remove confusion

File: test/RequestTest.php [hunk 2]
```diff
@@ -139,8 +139,9 @@ public function testInvalidParametersRequest()
 
   public function testRequestHeaders() {
     $req = new Castle_Request();
-    $raw = $req->send('GET', '/users');
+    $raw = $req->send('POST', '/events');
```
Comment (lluft) [inline test/RequestTest.php]: change to `track` endpoint `events` is legacy

**Comment by wallin** [comment]
@lluft Changes applied

**Pull request merged**